### PR TITLE
Add support to Rails 7 syntax

### LINKED ISSUE #20: syntax for declaring enums is incompatible with rails conventions
the documentation says to declare enums as I normally would

```ruby
class MyModel < ApplicationRecord
  enum model_type: [:default, :special]
end
```

but that is not how I would normally define enums :)
also, it's different from how the rails guide does it: https://guides.rubyonrails.org/active_record_querying.html#enums
it would be nice if this gem supported the list syntax also

### PR DESCRIPTION
**Pull request by guilhermearaujo**
This fixes #20

### CODE REVIEW

**Comment by rafasoares** [comment]
Hi @guilhermearaujo, thanks a lot for the PR!

Just a correction: 
`human_enum_value` and `self.human_enum_value` are exposed as convenience methods, so developers can fetch translated values without needing to set the property in the model instance.

**Review by rafasoares — CHANGES_REQUESTED** [review]

**Comment by rafasoares** [comment]
> > Hi @guilhermearaujo, thanks a lot for the PR!
> 
> > 
> 
> > Just a correction: `human_enum_value` and `self.human_enum_value` are exposed as convenience methods, so developers can fetch translated values without needing to set the property in the model instance.
> 
> 
> 
> Ok, I can undo that, no problem.
> 
> 
> 
> I'm not sure if I understood the use case for them. Tell me if this is correct:
> 
> `MyClass` has `my_enum`, and you would like to get the text for `:enum_value`, so, without any instance, you can call `MyClass.human_enum_value(:my_enum, :enum_value)`
> 
> Or if you have an instance, you could call `my_instance.human_enum_value(:my_enum, :enum_value)`
> 
> 
> 
> Is that what you mean? If so, one could easily also call `MyClass.human_my_enums[:enum_value]`

You're probably correct. It's been so long since the initial design of the gem that I don't really remember lol. 

Given my other comment, though, I think it's  still necessary to have this as a method in the module. Maybe we can follow the rails approach of prefixing the method with `_`?

What do you think about `_humanize_enum`? That way it's clear what its purpose is and that it's not part of the public API.

File: lib/human_enum.rb [hunk 1]
```diff
@@ -18,44 +18,36 @@ class Error < StandardError; end
...
-        human_enum name
+      if name.nil?
+        name = options.keys.reject { |key, _| key.start_with? '_'}.first
       end
-    end
 
-    def human_enum_value(enum_name, enum_value)
-      attributes_scope = "#{i18n_scope}.attributes"
-      enum_key = "#{enum_name.to_s.pluralize}.#{enum_value}"
+      humanize = lambda do |name, value, scope, ancestors|
```
Comment (guilhermearaujo) [inline lib/human_enum.rb:29]: Fixed by turning it into an anonymoys function

**Comment by guilhermearaujo** [comment]
@rafasoares is there anything else I could change?

**Comment by rafasoares** [comment]
Sorry @guilhermearaujo, I haven't had the time to properly review the new code. I'll do it in the next few days, promise :)

**Comment by rafasoares** [comment]
Superseded by #29

**Pull request closed without merging**